lib/utils/enum/enum_utils.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:logging/logging.dart';
+
+final Logger _logger = Logger('EnumUtils');
+
+/// Converts a string or int representation to its corresponding enum value.
+///
+/// Takes a dynamic [value] (expected to be a String or int) and attempts to match it
+/// against the [enumValues] list by comparing their name properties or by index.
+///
+/// If [value] is not a String or int, or if no matching enum is found, returns the
+/// [defaultValue].
+///
+/// Handles qualified enum names (e.g., 'EnumType.value') by extracting just the
+/// value part.
+///
+/// Example:
+/// ```
+/// final paymentType = parseEnum(
+///   value: 'send',
+///   enumValues: PaymentType.values,
+///   defaultValue: PaymentType.unknown
+/// );
+/// ```
+///
+/// [logFailures] controls whether parsing failures are logged (defaults to false).
+T parseEnum<T extends Enum>({
+  required dynamic value,
+  required List<T> enumValues,
+  required T defaultValue,
+  bool logFailures = false,
+}) {
+  // Handle numeric values (index-based)
+  if (value is int && value >= 0 && value < enumValues.length) {
+    return enumValues[value];
+  }
```

Comment:
Are enums normally stored as int or string? I wonder because inserting an enum in Rust could effect the representation here

Reply:
They are normally stored as string, with the enum `name`.

However, while it does not use this function, there is a case where the enum index(`int`) is used to store the enum.
`'appMode': appMode.index,`